hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/ha/TestStandbyBlockManagement.java

```diff
@@ -97,4 +97,58 @@ public void testInvalidateBlock() throws Exception {
     }
   }
 
+  /**
+   * Test Standby/Observer NameNode should not handle redundant replica block logic
+   * when set decrease replication.
+   * @throws Exception
+   */
+  @Test(timeout = 60000)
+  public void testNotHandleRedundantReplica() throws Exception {
+    Configuration conf = new Configuration();
+    HAUtil.setAllowStandbyReads(conf, true);
+    conf.setInt(DFSConfigKeys.DFS_HA_TAILEDITS_PERIOD_KEY, 1);
+
+    // Create HA Cluster.
+    try (MiniDFSCluster cluster = new MiniDFSCluster.Builder(conf)
+        .nnTopology(MiniDFSNNTopology.simpleHATopology()).numDataNodes(10).build()) {
```

Comment:
Here num of DataNode is 10, is it necessary? I think it is enough to set 4, what do you think about?

Reply:
yeah, here can set the number of datanode num to 4, which can also meet the test requirements.
I will update PR.